### PR TITLE
Remove resources from system addon manifests

### PR DESCRIPTION
**Pull request by vantuvt**
**What this PR does / why we need it**:
Update the setup-addon-manifests function to remove all fields contained
within and including the BEGIN_RESOURCES and END_RESOURCES markers, which have
been embedded as comments. The BEGIN_RESOURCE and END_RESOURCE markers were
chosen to be consistent with existing markers, such as, BEGIN_PROMETHEUS_TO_SD
and END_PROMETHEUS_TO_SD.

This logic is guarded by the ENABLE_SYSTEM_ADDON_VERTICAL_SCALING
environment variable, which is set to false by default rendering this change
benign. The removal of resource-related fields is necessary to enable
vertical scaling of system pods. Otherwise, the addon manager will undo
any changes specified by a scaler whenever it reconciles an addon's
definition.

Note that not all system manifests have been updated. Additional
follow-up changes may be needed pending further testing.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2042,6 +2042,17 @@ function setup-addon-manifests {
       copy-manifests "${psp_dir}" "${dst_dir}"
     fi
   fi
+
+  # If system addon vertical scaling is enabled, remove resource fields from the
+  # manifest.
+  if [[ "${ENABLE_SYSTEM_ADDON_VERTICAL_SCALING:-false}" == "true" ]]; then
+    # TODO: Check if there are any intermediate yaml files that need to be
+    # updated and are not caught by the regex passed to find.
+    echo DEBUG "${1:-}":"${2:-}":"${3:-}"
```
Comment (kawych) [inline cluster/gce/gci/configure-helper.sh]: If you want to log this, can you add some additional text to make it clearer?

**Comment by vantuvt** [comment]
I apologize if the way I've made the corrections doesn't follow best practices. I'm still learning what the proper workflow is for working with Github and the Kubernetes project. Please feel free to point me to docs describing best practices.

@kawych - Thank you for catching the debug logging. That was only intended for dev testing purposes. It has been removed.

I've also removed the changes to dashboard-controller.yaml as it was not necessary.

**Comment by vantuvt** [comment]
Friendly ping @bowei .

**Review by MrHohn — APPROVED** [review]
/approve
for dns-horizontal-autoscaler.

**Comment by vantuvt** [comment]
Friendly ping.

@nicksardo - Can you please take a look at cluster/addons/cluster-loadbalancing/glbc/default-svc-controller.yaml?

@bowei, @mwielgus - Can one of you please take a look at cluster/gce/gci/configure-helper.sh?


Thanks,
Van

**Comment by vantuvt** [comment]
This PR is unlikely to be needed. Closing.

**Pull request closed without merging**